lib/web_ui/lib/src/engine/canvaskit/surface.dart

```diff
@@ -47,11 +49,19 @@ class SurfaceFrame {
 /// The underlying representation is a [CkSurface], which can be reused by
 /// successive frames if they are the same size. Otherwise, a new [CkSurface] is
 /// created.
-class Surface {
-  Surface();
+class Surface extends OverlayCanvas {
+  Surface({this.isRenderCanvas = false})
+      : useOffscreenCanvas =
+            Surface.offscreenCanvasSupported && !isRenderCanvas;
 
   CkSurface? _surface;
 
+  /// Whether or not to use an `OffscreenCanvas` to back this [Surface].
+  final bool useOffscreenCanvas;
+
+  /// If `true`, this [Surface] is used for rendering.
```

Comment:
Does "render" here mean "display", i.e. it's shown to the user? Surfaces are always used for some kind of rendering, so it's unclear what it means for it to not be a render canvas.

Reply:
Done. Also, renamed `OverlayCanvas` to `DisplayCanvas` to make it clearer what these canvases are used for.